src/pages/index.js

```diff
@@ -67,250 +74,229 @@ const cardList = new Section({
 
-function renderCard(individualCard) {
-    const cardView = getCardView(individualCard);
-  //const card = new Card(data, cardSelector)
-  // container.prepend(card.getElementView(cardElement));
-  cardList.addItem(cardView)
-}
+const changeProfilePicPopup = new PopupWithForm({
+  modalSelector: ".modal_type_change_profile_pic-js",
+  handleFormSubmit: handleChangeAvatarSubmit
+})
+changeProfilePicPopup.setEventListeners()
 
 
 
+//FUNCTIONS
+function renderCard(individualCard) {
+  const cardView = getCardView(individualCard);
+  cardList.addItem(cardView);
+}
 
 //get a clone of card template and add the new source img and text it takes each obj iteration as cardData
 function getCardView(cardData) {
-  const card = new Card(cardData, '.card-template', handleImageClick)
-  return card.getElementView()
+  const card = new Card(cardData, ".card-template", handleImageClick, handleDeleteButton, handleCardLikeClick);
+  return card.getElementView();
+}
+
+//HANDLER FUNCTIONS
+//passing the whole card object
+function handleCardLikeClick(card) {
+  if (!card.isLiked) {
+    api.addTheCardLikeState(card._id).then(() => {
+      card.isLiked = true;
+      card.toggleLikeButton()
+    }).catch(console.error);
+  } else if (card.isLiked) {
+    api.deleteTheCardLikeState(card._id).then(() => {
+      card.isLiked = false;
+      card.toggleLikeButton()
+    }).catch(console.error);
+  }
 
 }
 
+
 //handler to set the text inside the edit  take out the add part of the function
 function handleEditButton() {
-    const userData = userInfo.getUserInfo()
-
+  const userData = userInfo.getUserInfo();
   //turn it into an arrow func prevent default and rearrange the value and text content and close the modal
   profileTitleInput.value = userData.userName;
-  profileDescriptionInput.value = userData.userDescription//profileDescription.textContent;
-
+  profileDescriptionInput.value = userData.userDescription; //profileDescription.textContent;
   // profileEditModal.classList.add("modal_opened");
-  editProfileForm.open()
- // openPopUp(profileEditModal);
-  // use .open() method of popup with form
+  editProfileForm.open();
 
 }
 
 //handler to set the text inside the edit
 function handleAddButton() {
-  //profileTitleInput.value = profileTitle.textContent;
-  // profileDescriptionInput.value = profileDescription.textContent;
-  // above are not needed for that modal
- // openPopUp(cardAddModal);
   addCardForm.open();
 }
 
 //handler function to update text inside edit profile
 function handleProfileEditSubmit(profileData) {
-  // event.preventDefault();
- //setUser info goes here
-  // set the instance method from UserInfo setuserinfo() for the code below
-  //profileTitle.textContent = profileData.name;
-  //profileDescription.textContent = profileData.description;
-  userInfo.setUserInfo(profileData)
-  editProfileForm.close();
-  //closePopUp(profileEditModal);
-}
+  editProfileForm.setLoadingButtonText(true)
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```